snuba/consumers/consumer.py

```diff
@@ -522,6 +642,8 @@ def __init__(
         input_block_size: Optional[int],
         output_block_size: Optional[int],
         metrics: MetricsBackend,
+        producer: Optional[AbstractProducer[KafkaPayload]],
+        topic: Optional[Topic],
```

Comment:
Maybe prefix these with `dead_letter` or something to make it explicit. Its possible that in the future we want to produce something on another topic which is not related to dead letter.